google/cloud/storage/fileio.py

```diff
@@ -394,12 +406,14 @@ def tell(self):
         return self._buffer.tell() + len(self._buffer)
 
     def flush(self):
-        if self._text_mode:
+        if self._ignore_flush:
```

Comment:
Can we refactor to avoid the early return?  E.g.:

```python
    def flush(self):
        if not self._ignore_flush:
            raise io.UnsupportedOperation(
                "Cannot flush without finalizing upload. Use close() instead, or "
                "set ignore_flush=True when constructing this class (see "
                "docstring)."
            )
```

Reply:
done